src/prognostic_equations/implicit/implicit_tendency.jl

```diff
@@ -9,6 +9,16 @@ NVTX.@annotate function implicit_tendency!(Yₜ, Y, p, t)
     Yₜ .= zero(eltype(Yₜ))
     Fields.bycolumn(axes(Y.c)) do colidx
         implicit_vertical_advection_tendency!(Yₜ, Y, p, t, colidx)
+        if p.atmos.diff_mode == Implicit()
```

Comment:
What's the difference between this and `p.atmos.diff_mode isa Implicit`?

Reply:
None; since `Implicit` is a singleton type, the assembly code should be identical. I just thought `==` is more readable/common than `isa`.